### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by andriy-shymkiv**
[DEMO LINK](https://andriy-shymkiv.github.io/react_people-table-basics/)

### CODE REVIEW

File: src/components/PeopleTable/PeopleTable.tsx [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    try {
+      getPeopleFromServer();
+    } catch {
+      setError(true);
+    }
+  }, []);
+
+  const findParentByName = (parentName: string | null) => {
+    return people?.find(
+      person => person.name === parentName || person.name === parentName,
```
Comment (iryna-lelyk) [inline src/components/PeopleTable/PeopleTable.tsx:32]:
```suggestion
      person => person.name === parentName,
```

File: src/components/PeopleTable/PeopleTable.tsx [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+      <div className="block">
+        <div className="box table-container">
+          {loader ? (
+            <Loader />
+          ) : (
+            <table
+              data-cy="peopleTable"
+              className="table is-striped is-hoverable is-narrow is-fullwidth"
+            >
```
Comment (iryna-lelyk) [inline src/components/PeopleTable/PeopleTable.tsx:54]: When you show your table, you should also check if it has any people in it. So better to separate loader and table with people
```suggestion
          {loader && (
            <Loader />
          )}
          
          {!people?.length && (
                  <p data-cy="noPeopleMessage">
                    There are no people on the server
                  </p>
                )}
          
          {!loader && people.length > 0 && (
            <table
              data-cy="peopleTable"
              className="table is-striped is-hoverable is-narrow is-fullwidth"
            >
```

File: src/components/PeopleTable/PeopleTable.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+                  <th>Father</th>
+                </tr>
+              </thead>
+
+              <tbody>
+                {people?.length === 0 && (
+                  <p data-cy="noPeopleMessage">
+                    There are no people on the server
+                  </p>
+                )}
```
Comment (iryna-lelyk) [inline src/components/PeopleTable/PeopleTable.tsx:71]: Better to move it outside of your table

File: src/components/PersonLink/PersonLink.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+      data-cy="person"
+      key={person.slug}
+      className={classNames({
+        'has-background-warning': slug === person.slug,
+      })}
+    >
+      <td>
+        <Link
+          to={`/people/${person.slug}`}
+          className={`${person.sex === 'f' && 'has-text-danger'}`}
```
Comment (iryna-lelyk) [inline src/components/PersonLink/PersonLink.tsx:29]: Use here `classNames`.

File: src/components/PersonLink/PersonLink.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+            <Link
+              className="has-text-danger"
+              to={`/people/${mother.slug}`}
+            >
+              {mother.name}
+            </Link>
+          )
+          : `${person.motherName
+            ? `${person.motherName}`
+            : '-'}`}
```
Comment (iryna-lelyk) [inline src/components/PersonLink/PersonLink.tsx:51]: It's hard to understand several ternar operators. Try to simplify it.
```suggestion
          : `${person.motherName}` || '-'
           }
```

**Review by mykola-koval — APPROVED** [review]

**Pull request closed without merging**